### PR TITLE
Add gitignore

### PR DESCRIPTION
**Pull request by ludicast**
## Problem

We need gitignore values added

## Solution

This adds the git ignore, importing the `ignore` package (which is used in all other nx-plugins for setting up ignore files). We also are already importing it.

### Useful documentation

- this is mostly a clone of how gitignores are done throughout the nrwl monorepo: https://github.com/nrwl/nx/search?q=gitignore

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -53,6 +53,7 @@
         "functional/functional-parameters": "off",
         "functional/no-expression-statement": "off",
         "functional/no-return-void": "off",
+        "functional/no-let": "off",
```
Comment (ludicast) [inline .eslintrc.json:56]: this is mostly covered by the `prefer-const` rule that we are following (confirmed this is working in the repo already).  Banishing `let`s completely makes the coding more academic than practical (e.g.we would need to `reduce` the lines of our gitignore file or something crazy like that :))

File: local-test.sh [hunk 2]
```diff
@@ -60,7 +60,6 @@ stash_workspace_changes_if_requested
 yarn unlink $PLUGIN_NPM_NAME
 yarn link $PLUGIN_NPM_NAME
 yarn nx generate $PLUGIN_NPM_NAME:project $app-e2e --project $app
-yarn playwright install --with-deps
```
Comment (ludicast) [inline local-test.sh:63]: no longer needed, because playwright automatically installs on init now/

File: .eslintrc.json [hunk 3]
```diff
@@ -77,7 +78,6 @@
     {
       "files": ["*.spec.*"],
       "rules": {
-        "functional/no-return-void": "off",
```
Comment (ludicast) [inline .eslintrc.json:80]: rule already applied to `ts` files above

File: package.json [hunk 4]
```diff
@@ -50,6 +50,7 @@
     "eslint-plugin-simple-import-sort": "^7.0.0",
     "eslint-plugin-unused-imports": "^2.0.0",
     "husky": "^8.0.1",
+    "ignore": "^5.2.0",
```
Comment (ludicast) [inline package.json:53]: doesn't affect yarn.lock because it's already imported

File: package.json [hunk 5]
```diff
@@ -50,7 +50,9 @@
     "eslint-plugin-simple-import-sort": "^7.0.0",
     "eslint-plugin-unused-imports": "^2.0.0",
     "husky": "^8.0.1",
-    "jest": "27.5.1",
+    "ignore": "^5.2.0",
+    "jest": "2
```
Comment (andreacaldera) [inline package.json:54]: Looks like you added a return character by mistake here?

File: .eslintrc.json (hunk 1 above)
Comment (andreacaldera) [reply .eslintrc.json:56]: I'd still like to keep the rule as `let` could lead to poorly written and hard to follow code. I'll suggest something below to use instead to save you from re-typing that ;-)

File: packages/nx-playwright/src/generators/init/lib/add-git-ignore-entry.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import { Tree } from '@nrwl/devkit';
```
Comment (andreacaldera) [inline packages/nx-playwright/src/generators/init/lib/add-git-ignore-entry.ts:1]:
```suggestion
import { Tree } from '@nrwl/devkit';
import ignore from 'ignore';

export const addGitIgnoreEntry = (host: Tree) => {
  const content = host.exists('.gitignore') ? host.read('.gitignore', 'utf-8').trimEnd() : '';

  const ig = ignore();
  ig.add(host.read('.gitignore', 'utf-8'));

  const addIgnore = (existingIgnore: string, ignoreToAdd: string) => {
    return ig.ignores(existingIgnore) ? '' : ignoreToAdd;
  };

  const resultsIgnore = addIgnore('app/example/test-results', `\n\n# Playwright\n**/test-results/`);
  const reportsIgnore = addIgnore('app/example/playwright-report', `\n**/playwright-report/`);
  const cacheIgnore = addIgnore('app/example/playwright/.cache', '\n**/playwright/.cache/\n');

  host.write('.gitignore', `${content}${resultsIgnore}${reportsIgnore}${cacheIgnore}`);
};

```

File: package.json (hunk 5 above)
Comment (ludicast) [reply package.json]: thanks :man_facepalming:

File: packages/nx-playwright/src/generators/init/lib/add-git-ignore-entry.ts (hunk 6 above)
Comment (ludicast) [reply packages/nx-playwright/src/generators/init/lib/add-git-ignore-entry.ts:1]: thanks, very nice cleanup @andreacaldera!

File: .eslintrc.json (hunk 1 above)
Comment (andreacaldera) [reply .eslintrc.json:56]: Could we keep let disabled then and remove this line? :-)

Comment (ludicast) [reply .eslintrc.json:56]: sure.

in part I feel dis-allowing `let`s is extreme. but I also agreee that code with `const`s is sometimes cleaner and more correct (albeit sometimes littered with `reduce`s and splatting).

File: packages/nx-playwright/src/generators/init/lib/add-git-ignore-entry.ts (hunk 6 above)
Comment (ludicast) [reply packages/nx-playwright/src/generators/init/lib/add-git-ignore-entry.ts:1]: since your suggested change added code rather than replaced it, when I removed the extra function I accidentally removed yours @andreacaldera .  re-added & thanks again

**Comment by andreacaldera** [comment]
> @andreacaldera - added test coverage. also, think I can have auto-running workflows, definitely makes a barrier to contributing (says the restriction is for first-time contributors, which I am not :) ).

Thanks and agreed, this needs fixing as I'm already tired of pressing the approve and run button ;-)

**Comment by ludicast** [comment]
cool - running seems fixed @andreacaldera

**Review by andreacaldera — APPROVED** [review]
